data_structures/src/transaction.rs

```diff
@@ -161,9 +170,19 @@ impl VTTransaction {
         VTTransaction { body, signatures }
     }
 
-    /// Returns the byte size that a transaction will have on the wire
+    /// Returns the size that a value transfer transaction will have on the wire
+    #[allow(clippy::cast_possible_truncation)]
     pub fn size(&self) -> u32 {
```

Comment:
This is not calculating the serialized transaction size now:

```suggestion
    /// Returns the weight that a value transfer transaction. This is the weight that will be used to calculate how many transactions can fit inside one block
    pub fn weight(&self) -> u32 {
```